boards/shields/nrf21540_ek/Kconfig.defconfig

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2021, Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if SHIELD_NRF21540_EK
+
+config SPI
+	default y
```

Comment:
Everything is right here.

Reply:
OK, so I dug into this.  I agree this is the technique used in most shields, but on first look it does seem better to use `select` in the way `depends_on` is used, by adding it to the variable that's used as the external condition.

https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#select-statements seems to jump around with the main conclusion being "don't use `select`".

Is that your point @jfischer-no?  If so I'm fine with leaving this as-is.

Reply:
I understood that `default` is something that may be overwritten by user config, but `select` is not. Because of that using `default` here could result in invalid configuration for given shield.

Reply:
But this shield can work without SPI, right?  At least, nRF21540 can be operated in GPIO-only mode without SPI.  So `select` would not be what we want for that use case.

Reply:
Right, I didn't think about it. I agree select would be a poor choice for this shield.

Reply:
It is fine how it is. `select` is evil :-)

Reply:
@KAGA164 Where is the driver for the FEM? Is it completely configurable via devicetree?

Reply:
Currently there is no driver in the Zephyr for this FEM

Reply:
I am confused, then what is this shield support for?

Reply:
I think this is as a complement to the description of the nrf21540dk and there are users who create their own  driver for the FEM, it could be useful for them

Reply:
I agree with @KAGA164. It's useful for out-of-tree applications utilizing nRF21540. Moreover, BLE Controller module should use this devicetree description what is requested in #30708

Reply:
> I think this is as a complement to the description of the nrf21540dk and there are users who create their own driver for the FEM, it could be useful for them

That (regarding SPI) does not seem right to me. I do not think we should have a shield support for imaginary driver/hardware. At least all SPI relevant options and properties should be removed because there is no driver for that. Further, we need a test case for this shield.

Reply:
I can remove this configuration for the SPI. What do you mean by tests for the shield ?

Reply:
Is there is a purpose for this in the tree, a Bluetooth or 802154 sample that could use one of the FEM functionalities? In this sample, you could add an `extra_args: SHIELD=nrf21540ek` option to sample.yaml ,
```yaml
  sample.nrf21540ek:
    depends_on: arduino_gpio
    extra_args: SHIELD=nrf21540ek
    tags: shield
```